src/main/java/de/mediathekview/mserver/crawler/sr/tasks/SrTopicArchivePageTask.java

```diff
@@ -0,0 +1,67 @@
+package de.mediathekview.mserver.crawler.sr.tasks;
+
+import de.mediathekview.mserver.base.Consts;
+import de.mediathekview.mserver.crawler.basic.AbstractCrawler;
+import de.mediathekview.mserver.crawler.basic.AbstractDocumentTask;
+import de.mediathekview.mserver.crawler.basic.AbstractUrlTask;
+import de.mediathekview.mserver.crawler.sr.SrTopicUrlDTO;
+import de.mediathekview.mserver.crawler.sr.SrConstants;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+
+public class SrTopicArchivePageTask extends AbstractDocumentTask<SrTopicUrlDTO, SrTopicUrlDTO> {
+
+  private static final Logger LOG = LogManager.getLogger(SrTopicArchivePageTask.class);
+  
+  private static final String NEXT_PAGE_SELECTOR = "div.pagination__item > a[title*=weiter]";
+  private static final String SHOW_LINK_SELECTOR = "h3.teaser__text__header a";
+  
+  public SrTopicArchivePageTask(final AbstractCrawler aCrawler,
+      final ConcurrentLinkedQueue<SrTopicUrlDTO> aUrlToCrawlDTOs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDTOs" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)